### PR TITLE
feat(dart_frog_cli): press R to reload

### PR DESCRIPTION
**Pull request by renancaraujo**
## Status

**READY**

## Description

Proposal for support R to hot reload just like Flutter does. 

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart [hunk 1]
```diff
@@ -156,12 +161,19 @@ class DevServerRunner {
...
   }
 
-  Future<void> _reload() async {
-    logger.detail('[codegen] reloading...');
+  Future<void> _reload([bool verbose = false]) async {
+    final void Function(String) log;
+    if (verbose) {
+      log = logger.info;
+    } else {
+      log = logger.detail;
```
Comment (alestiago) [inline packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:169]: **Nit & Question**: Why are we preferring to use `logger.info` or `logger.detail` over `logger.progress` to indicate the progress of the reload?

File: packages/dart_frog_cli/lib/src/commands/dev/dev.dart [hunk 2]
```diff
@@ -45,13 +47,60 @@ class DevCommand extends DartFrogCommand {
...
+    if (!_stdin.hasTerminal) return;
+
+    // listen for the R key
+    _stdin
+      ..echoMode = false
+      ..lineMode = false;
+
+    _stdinSubscription = _stdin.listen(
+      (event) {
+        if (event.length == 1 && event.first == 82) {
```
Comment (alestiago) [inline packages/dart_frog_cli/lib/src/commands/dev/dev.dart:73]:
```suggestion
        if (event.length == 1 && event.first == 'R'.codeUnitAt(0)) {
```

**Review by alestiago — APPROVED** [review]
LGTM! 💙

File: packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart (hunk 1 above)
Comment (renancaraujo) [reply packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:169]: That's an idea for a further PR

Comment (alestiago) [reply packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart:169]: Okay, sounds good to me. I thought it was intentional.

**Review by wolfenrain — APPROVED** [review]

**Review by alestiago — APPROVED** [review]

**Pull request merged**